frontend/src/patched/JIT.js

```diff
@@ -2561,40 +2604,46 @@ Extras.Classes.Navigation = new Class({
     currentPos.y += oy;
     var x = currentPos.x - thispos.x,
         y = currentPos.y - thispos.y;
-    
+
     // START METAMAPS CODE
-    Metamaps.Mouse.changeInX = x;
-    Metamaps.Mouse.changeInY = y;
+    this.changeInX = x;
+    this.changeInY = y;
+    this.didPan = true;
     // END METAMAPS CODE
-    
+
     this.pos = currentPos;
     this.canvas.translate(x * 1/sx, y * 1/sy);
 
     // START METAMAPS CODE
-      jQuery(document).trigger(Metamaps.JIT.events.pan);
+    this.config.onPan();
     // END METAMAPS CODE
   },
-  
+
   onMouseUp: function(e, win, eventInfo, isRightClick) {
     if(!this.config.panning) return;
     this.pressed = false;
-    
+
     // START METAMAPS CODE
-    if (Metamaps.Mouse.didPan) Metamaps.JIT.SmoothPanning();
-    
-    
+    if (this.boxStartCoordinates && this.boxEndCoordinates) {
+      this.zoomToBox();
+      this.config.onZoom(e);
+    }
+    else if (this.didPan) {
+      this.smoothPanning();
+    }
+    this.boxStartCoordinates = null;
+    this.boxEndCoordinates = null;
     // END METAMAPS CODE
-    
   },
-  
+
   // START METAMAPS CODE
   onTouchStart: function(e, win, eventInfo) {
     if(!this.config.panning) return;
-    Metamaps.Mouse.changeInX = 0;
-    Metamaps.Mouse.changeInY = 0;
+    this.changeInX = 0;
+    this.changeInY = 0;
     if((this.config.panning == 'avoid nodes' && eventInfo.getNode()) || eventInfo.getEdge()) return;
     this.pressed = true;
-    Metamaps.Mouse.didPan = false;
+    this.didPan = false;
```

Comment:
Awesome, does this mean we are closer to removing some of the Metamaps.Mouse stuff?

I think that's a good change. We have state in a lot of places. i.e. JIT, Metamaps.DataModel, Metamaps.Mouse, etc.

If JIT's state is explicitly held by JIT, that's a good thing.

Reply:
that's the goal yeah. so this cuts out some of metamaps.mouse